src/other/index.jl

```diff
@@ -218,9 +218,26 @@ end
     if notidx.skip isa AbstractVector 
         toskip = filter(t -> haskey(x, t), notidx.skip)
         return setdiff(1:length(x), toskip)
+    elseif notidx.skip isa Between
+        if haskey(x, notidx.skip.first) && haskey(x, notidx.skip.last)
+            toskip = x[notidx.first]:x[notidx.last]
+            return setdiff(1:length(x), toskip)
+        elseif !(haskey(x, notidx.skip.first)) && !(haskey(x, notidx.skip.last))
+            return 1:length(x)
```

Comment:
why do you think it should not error?

Reply:
I would be fine having this error. 

Here is my reasoning for the current behavior. Imagine a dataframe with infinite columns, and a contiguous set of columns comprises your data frame. When you do `Not(Between(:x, :z))`, either 

1. `:x` and `:z` are both to the left of your columns of interest. Then no error, do nothing
2. `:x` and `:z` are both to the right of your columns of interest. Then no error, do nothing. 
3. `:x` is on the left of your columns of interest, `:z` is on the right. In which case you should delete all columns. 

This PR, currently, only assumes options 2 and 3.

Reply:
But if `!(haskey(x, notidx.skip.first)) && !(haskey(x, notidx.skip.last))` is true then you are not sure if

`:x` is on the left and `:z` is on the right (and do what you proposed)

or

`:z` is on the left and `:x` is on the right (and then exactly the opposite should happen)

That is why I felt it should also error.

If we agree then `Between` should be removed from the list in my main comment.